hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/HazelcastWriters.java

```diff
@@ -152,6 +159,45 @@ public static ProcessorMetaSupplier writeListSupplier(@Nonnull String name, @Nul
         ));
     }
 
+    public static ProcessorMetaSupplier writeObservableSupplier(@Nonnull String name) {
+        return new ProcessorMetaSupplier() {
+
+            private JetInstance jet;
+
+            @Override
+            public void init(@Nonnull Context context) {
+                this.jet = context.jetInstance();
+            }
+
+            @Override
+            public void close(@Nullable Throwable error) {
+                ObservableRepository.completeObservable(name, error, jet, System::currentTimeMillis);
+                //todo: force single event?
```

Comment:
This method is not called exactly-once:
- If the coordinator dies and job doesn't restart (typical for batch jobs), it's not called at all
- If the job restarts, it's called after each execution.
We should tie the observable lifecycle to that of JobResult, as I suggested elsewhere.

Reply:
Absolutely correct, this `close` method based solution is no good. Have added a test to prove it (`ObservableResultsTest.streamJobRestart`). I have reverted to the solution I had before polluting PMS was raised as a concern. 

@cangencer I don't see any alternative to polluting the PMS interface, but this time I tried to do it in a more generic, hopefully reusable way which hopefully makes some sense to have and you can  accept it. Pls. take a look and let's see. I would gladly not do it, but I see no way around it. I need some reliable way to figure out what Observables are owned by a DAG. I also contemplated storing something during DAG construction, but I guess I can't assume that the DAG is not coming from some external source, already prebuilt...